### PR TITLE
Use the precise number of approvals when constructing RawOrgin::Members

### LINKED ISSUE #9604: Use `yes_votes` instead of `voting.threshold` when constructing `RawOrigin::Members` in pallet-collective
https://github.com/paritytech/substrate/blob/9d867b6b4dd099ecef7bed1a1ff5a3441f41d684/frame/collective/src/lib.rs#L774

### Advantage

The drawback of using `voting.threshold` here is that each runtime can have its own threshold rule but this parameter now is not dynamic and hardcoded on the JS end, rendering the parameter of `threshold` submitted from the JS apps incorrect when a runtime uses a non-default value `1/2`.

https://github.com/polkadot-js/apps/blob/777584eaad43c3ae6a46799e2fb7557221e6dc38/packages/apps-config/src/api/params/proposalThresholds.ts#L18-L23

Using `yes_notes` does not resolve the UI issue at all, but allows us to execute the call anyhow even the UI is not right. Say, a chain uses 3/4 for `ExternalMajorityOrigin` and the count of council members is 11, when a member clicks the `Propose External` button at https://polkadot.js.org/apps/, UI shows a threshold of 6 (the correct display is 9). 6 members can easily click the `vote` button to approve, 3 members can still approve this by using the `extrinsics` interface later. Eventually, the number of approvals becomes 9, the proposal should be approved, but due to using the initial incorrect threshold is constructed in `RawOrigin::Members(6, 11)`, the whole process simply fails, that does not make sense as we do have enough votes. Using `yes_notes` will help this.

### Disadvantage

I can't think of any disadvantages, using `yes_votes` does not change the semantics and it's just more accurate IMO.

### PR DESCRIPTION
**Pull request by liuchengxu**
Close #9604

### CODE REVIEW

**Comment by liuchengxu** [comment]
Can you please take a look at this PR? @shawntabrizi @thiolliere

**Comment by shawntabrizi** [comment]
can you please write a test?

**Comment by shawntabrizi** [comment]
Okay, this makes sense to me. Can you please write a test?

**Comment by liuchengxu** [comment]
@shawntabrizi Added a test in https://github.com/liuchengxu/substrate/commit/118c272d5917ff705f461b71febcbee458963790. I also separated out `tests.rs` and did some refactoring, but should be easy to review by each commit.

**Comment by thiolliere** [comment]
maybe you need to merge master to fix CI

File: primitives/state-machine/src/lib.rs [hunk 1]
```diff
@@ -1579,7 +1579,6 @@ mod tests {
 		let mut seed = [0; 16];
 		for i in 0..50u32 {
 			let mut child_infos = Vec::new();
-			&seed[0..4].copy_from_slice(&i.to_be_bytes()[..]);
```
Comment (thiolliere) [inline primitives/state-machine/src/lib.rs:1582]: the correct fix is probably https://github.com/paritytech/substrate/pull/9747

Comment (liuchengxu) [reply primitives/state-machine/src/lib.rs:1582]: Will do once it's merged into master.

File: frame/collective/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,981 @@
...
+fn motions_approval_with_enought_votes_and_lower_voting_threshold_works() {
+	new_test_ext().execute_with(|| {
+		let proposal = Call::Democracy(mock_democracy::Call::external_propose_majority());
+		let proposal_len: u32 = proposal.using_encoded(|p| p.len() as u32);
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		let hash: H256 = proposal.blake2_256().into();
+		// The voting threshold is 2, but the required votes for `ExternalMajorityOrigin` is 3.
+		// The proposal will be executed regardless of the voting threshold
+		// as long as we have enough yes votes.
+		assert_ok!(Collective::propose(
```
Comment (thiolliere) [inline frame/collective/src/tests.rs:787]: maybe we can also do after a propose closed with 2 votes yes, to ensure that when `vote == threshold` it is indeed failing.

just to ensure that if somebody change the configuration by mistake then we don't end up with a test which is always successful.

**Review by thiolliere — COMMENTED** [review]
makes sense to me too.

**Review by thiolliere — APPROVED** [review]
looks good to me,

* [x] maybe we should implement this suggestion: https://github.com/paritytech/substrate/pull/9647/files#r706215564
* [ ] and we should revert this change https://github.com/paritytech/substrate/pull/9647/files#r706082154

File: frame/collective/src/tests.rs (hunk 2 above)
Comment (liuchengxu) [reply frame/collective/src/tests.rs:789]: addressed in https://github.com/paritytech/substrate/pull/9647/commits/e81e06ac9e2cee86a894180e188e696e08fdbc98

**Comment by liuchengxu** [comment]
>and we should revert this change https://github.com/paritytech/substrate/pull/9647/files#r706082154

@thiolliere Not sure I fully understand this, do you mean we should use the hardcoded proposal hash instead of using the variable `hash`?

**Comment by thiolliere** [comment]
> > and we should revert this change https://github.com/paritytech/substrate/pull/9647/files#r706082154
> 
> @thiolliere Not sure I fully understand this, do you mean we should use the hardcoded proposal hash instead of using the variable `hash`?

it was about the commit to try to fix CI. But master has been merged with the fix. so all good now.

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**